src/trainerai/pick_command.c

```diff
@@ -8,58 +8,1279 @@
+         return ABILITY_NONE;
+     } //Really should also check lightningrod, storm drain, earth eat, and dry skin -- TODO I guess.
+ 
+     // If our ability absorbs the type of the last move that hit us, do not switch.
+     if (battleCtx->battlemon[battler].ability == checkAbility) {
+         return FALSE;
+     }
+ 
+
+ 
+     aiSlot1 = battler;
+     if ( battleType & (BATTLE_TYPE_MULTI | BATTLE_TYPE_TAG)) {
+         aiSlot2 = aiSlot1;
+     } else {
+         aiSlot2 = BATTLER_ALLY(battler);
+     }      //the idea with this logic is that we double-check our own slot twice if multi battle, but for regular double battles with 1 trainer on either side, we need to check both slots
+
+
+     start = 0;
+     end = Battle_GetClientPartySize(battleSys, battler);
+ 
+     // Check each Pokemon on the bench for one which has an ability that absorbs
+     // the last move that was used.
+     for (i = start; i < end; i++) {
+         mon = Battle_GetClientPartyMon(battleSys, battler, i);
+ 
+         if (GetMonData(mon, MON_DATA_HP, 0) != 0 &&
+         GetMonData(mon, MON_DATA_SPECIES_OR_EGG, 0) != 0 &&
+         GetMonData(mon, MON_DATA_SPECIES_OR_EGG, 0) != 494 
+         && i != battleCtx->sel_mons_no[aiSlot1]
+         && i != battleCtx->sel_mons_no[aiSlot2]
+         && i != battleCtx->aiSwitchedPartySlot[aiSlot1]
+         && i != battleCtx->aiSwitchedPartySlot[aiSlot2]) {
+             ability = GetMonData(mon, MON_DATA_ABILITY, NULL);
+ 
+             // Switch to a matching Pokemon 50% of the time.
+             if (checkAbility == ability && (BattleRand(battleSys) & 1)) {
+                 battleCtx->aiSwitchedPartySlot[battler] = i;
+                 return TRUE;
+             }
+         }
+     }
+ 
+     return FALSE;
+ }
+
+
+ 
+/**
+ * @brief Check if an AI's battler has a super-effective move against either of the
+ * opponent's Pokemon.
+ *
+ * @param battleSys
+ * @param battleCtx
+ * @param battler   The AI's battler.
+ * @param flag      If TRUE, will always return TRUE if the AI's battler has a super-
+ *                  effective move. If FALSE, returns TRUE 90% of the time for either
+ *                  target against which the battler has a super-effective move.
+ * @return TRUE if the AI's battler has a super-effective move.
+ */
+BOOL AI_HasSuperEffectiveMove(struct BattleSystem *battleSys, struct BattleStruct *battleCtx, int battler, BOOL flag)
+{
+    int i;
+    u32 effectiveness;
+    u8 defender;
+    u8 oppositeSlot;
+    u16 move;
+    int type;
+    int battleType = BattleTypeGet(battleSys);
+    // Look at the slot directly across from us on the opposite side. i.e.,
+    // AI slot 1 looks at player slot 1, AI slot 2 looks at player slot 2
+    //oppositeSlot = BattleSystem_BattlerSlot(battleSys, battler) ^ 1;
+    //defender = BattleSystem_BattlerOfType(battleSys, oppositeSlot);
+
+    defender = BATTLER_OPP(battler);
+    
+    //if ((battleCtx->battlersSwitchingMask & FlagIndex(defender)) == FALSE) { 
```

Comment:
Remove comment that looks like code to reduce mental load. Same for the following comments.